File: TransCoder_tokenized_test_set_functions/python/DELETE_ARRAY_ELEMENTS_WHICH_ARE_SMALLER_THAN_NEXT_OR_BECOME_SMALLER.py
def deleteElements ( arr, n, k ) :
    st = [ ]
    st . append ( arr [ 0 ] )
    top = 0
    count = 0
    for i in range ( 1, n ) :
        while ( len ( st ) != 0 and count < k and st [ top ] < arr [ i ] ) :
            st . pop ( )
            count += 1
            top -= 1
        st . append ( arr [ i ] )
        top += 1
    for i in range ( 0, len ( st ) ) :
        print ( st [ i ], " ", end = "" )
    

File: TransCoder_tokenized_test_set_functions/java/DELETE_ARRAY_ELEMENTS_WHICH_ARE_SMALLER_THAN_NEXT_OR_BECOME_SMALLER.java
static void deleteElements ( int arr [ ], int n, int k ) {
  Stack < Integer > s = new Stack < > ( );
  s . push ( arr [ 0 ] );
  int count = 0;
  for ( int i = 1;
  i < n;
  i ++ ) {
    while ( ! s . empty ( ) && s . peek ( ) < arr [ i ] && count < k ) {
      s . pop ( );
      count ++;
    }
    s . push ( arr [ i ] );
  }
  int m = s . size ( );
  Integer [ ] v = new Integer [ m ];
  while ( ! s . empty ( ) ) {
    v [ -- m ] = s . peek ( );
    s . pop ( );
  }
  for ( Integer x : v ) {
    System . out . print ( x + " " );
  };
  System . out . println ( "" );
}